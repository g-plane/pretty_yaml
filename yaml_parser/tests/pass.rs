use insta::{assert_snapshot, glob, Settings};
use std::{fs, path::Path};

#[test]
fn pass_snapshot() {
    glob!("pass/*.yaml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let tree = match yaml_parser::parse(&input) {
            Ok(tree) => tree,
            Err(err) => panic!("failed to parse '{}': {err}", path.display()),
        };
        assert_eq!(
            tree.to_string(),
            input,
            "syntax tree of '{}' does not match source",
            path.display()
        );

        build_settings(path).bind(|| {
            let name = path.file_stem().unwrap().to_str().unwrap();
            assert_snapshot!(name, format!("{tree:#?}"));
        });
    });
}

fn build_settings(path: &Path) -> Settings {
    let mut settings = Settings::clone_current();
    settings.set_snapshot_path(path.parent().unwrap());
    settings.remove_snapshot_suffix();
    settings.set_prepend_module_to_snapshot(false);
    settings.remove_input_file();
    settings.set_omit_expression(true);
    settings.remove_input_file();
    settings.remove_info();
    settings
}
