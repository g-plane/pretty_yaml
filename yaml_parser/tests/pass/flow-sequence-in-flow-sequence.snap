---
source: yaml_parser/tests/pass.rs
---
ROOT@0..12
  DOCUMENT@0..11
    BLOCK@0..11
      FLOW@0..11
        FLOW_SEQ@0..11
          L_BRACKET@0..1 "["
          FLOW_SEQ_ENTRIES@1..10
            FLOW_SEQ_ENTRY@1..2
              FLOW@1..2
                PLAIN_SCALAR@1..2 "a"
            COMMA@2..3 ","
            WHITESPACE@3..4 " "
            FLOW_SEQ_ENTRY@4..10
              FLOW@4..10
                FLOW_SEQ@4..10
                  L_BRACKET@4..5 "["
                  FLOW_SEQ_ENTRIES@5..9
                    FLOW_SEQ_ENTRY@5..6
                      FLOW@5..6
                        PLAIN_SCALAR@5..6 "b"
                    COMMA@6..7 ","
                    WHITESPACE@7..8 " "
                    FLOW_SEQ_ENTRY@8..9
                      FLOW@8..9
                        PLAIN_SCALAR@8..9 "c"
                  R_BRACKET@9..10 "]"
          R_BRACKET@10..11 "]"
  WHITESPACE@11..12 "\n"
